app/src/main/java/org/jellyfin/mobile/player/qualityoptions/QualityOptionsProvider.kt

```diff
@@ -0,0 +1,43 @@
+package org.jellyfin.mobile.player.qualityoptions
+
+class QualityOptionsProvider {
+
+    private val defaultQualityOptions = listOf(
+        QualityOption(maxHeight = 2160, bitrate = 120000000),
+        QualityOption(maxHeight = 2160, bitrate = 80000000),
+        QualityOption(maxHeight = 1080, bitrate = 60000000),
+        QualityOption(maxHeight = 1080, bitrate = 40000000),
+        QualityOption(maxHeight = 1080, bitrate = 20000000),
+        QualityOption(maxHeight = 1080, bitrate = 15000000),
+        QualityOption(maxHeight = 1080, bitrate = 10000000),
+        QualityOption(maxHeight = 720, bitrate = 8000000),
+        QualityOption(maxHeight = 720, bitrate = 6000000),
+        QualityOption(maxHeight = 720, bitrate = 4000000),
+        QualityOption(maxHeight = 480, bitrate = 3000000),
+        QualityOption(maxHeight = 480, bitrate = 1500000),
+        QualityOption(maxHeight = 480, bitrate = 720000),
+        QualityOption(maxHeight = 360, bitrate = 420000),
+        QualityOption(maxHeight = 0, bitrate = 0), // auto
+    )
+
+    @Suppress("MagicNumber")
+    fun getApplicableQualityOptions(videoWidth: Int, videoHeight: Int): List<QualityOption> {
+        // If the aspect ratio is less than 16/9 (1.77), set the width as if it were pillarboxed
+        // i.e. 4:3 1440x1080 -> 1920x1080
+        val maxAllowedWidth = when {
+            videoWidth / videoHeight < 16 / 9 -> videoHeight * (16 / 9)
```

Comment:
Probably makes sense to move `16/9` to a constant so it's pre-calculated on use (ASPECT_RATIO_16_9)

Reply:
The code was actually broken anyway, all of those values were `Ints`…
I fixed it and use a `Rational` to compare them now.